### PR TITLE
Add GitHub action for releasing CLI.

### PR DESCRIPTION
**Pull request by morgante**
This should help us to release faster with less toil.

### CODE REVIEW

**Comment by morgante** [comment]
@bharathkkb PTAL

File: .github/workflows/release-cli.yml [hunk 1]
```diff
@@ -0,0 +1,98 @@
+name: Release new CLI versions
+on:
+  push:
+    branches: 
+      - "master"
+    paths:
+      - "cli/Makefile"
+env:
+  CLI_BUCKET: gs://cft-cli
```
Comment (bharathkkb) [inline .github/workflows/release-cli.yml:9]: Are we using this in the Makefile? I also saw the same value being defined in the `cli/Makefile` as `BUCKET=gs://cft-cli`

File: .github/workflows/release-cli.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      - uses: GoogleCloudPlatform/github-actions/setup-gcloud@master
+        with:
+          version: "286.0.0"
+          service_account_key: ${{ secrets.GCP_SA_KEY }}
+
+      - name: Retrieve last released version
+        run: |-
+          LAST_VERSION=$(curl --silent "https://api.github.com/repos/GoogleCloudPlatform/cloud-foundation-toolkit/releases/latest" | \
+            grep '"tag_name":' |
+            sed -E 's/.*"([^"]+)".*/\1/')
```
Comment (bharathkkb) [inline .github/workflows/release-cli.yml:28]: Can we not use `jq` instead which maybe cleaner if all we need is the value of `tag_name` like we do for TF latest release here https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/1f47e570882acc69ee9f7ce6a9c6a12149d6f82e/.github/workflows/update-tooling.yml#L17

File: .github/workflows/release-cli.yml (hunk 1 above)
Comment (morgante) [reply .github/workflows/release-cli.yml:9]: Yeah, we have it set in both places. We could remove from Makefile eventually but for now I'm leaving as-is.

File: cli/cmd/version.go [hunk 3]
```diff
@@ -17,6 +19,6 @@ var versionCmd = &cobra.Command{
 example: CFT CLI version v1.0.0`,
 	Args: cobra.NoArgs,
 	Run: func(cmd *cobra.Command, args []string) {
-		cmd.Println("CFT CLI version", Version)
+		fmt.Println(Version)
```
Comment (katze120) [inline cli/cmd/version.go:22]: This change is not passing the [test case here](https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/master/cli/cmd/version_test.go#L16)

Comment (morgante) [reply cli/cmd/version.go:22]: Oops! Fixing: #701

**Pull request merged**